File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

let sumOfMultiples (numbers: int list) (upperBound: int): int = 
    [1..(upperBound - 1)]
    |> List.where (fun number -> List.exists (fun multiple -> number % multiple = 0) numbers)
    |> Seq.distinct
    |> List.ofSeq
    |> List.sum

File: scala/sum-of-multiples/src/main/scala/SumOfMultiples.scala
object SumOfMultiples {
  def sum(factors: Set[Int], limit: Int): Int = {
    def getMultiples(factor: Int, limit: Int): Seq[Int] = {
      for(i <- Range(0, limit) if  i % factor == 0) yield i;
    } 

    factors
      .flatMap(f => getMultiples(f, limit))
      .sum
  }    
}

